### PR TITLE
Symintify guards.cpp

### PR DESCRIPTION
**Pull request by soulitzer**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #110529
* #110528
* #109171
* #110372
* __->__ #110371
* #110370
* #110369
* #110044

Separating this out so we can check perf more easily

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: torch/csrc/dynamo/guards.cpp [hunk 1]
```diff
@@ -265,14 +270,15 @@ static int TensorGuards_init(
...
+    std::vector<std::optional<c10::SymInt>> tensor_dims_size =
         per_tensor_dynamic_dims_sizes.empty()
-        ? wrapIntegersInOptional(tensor.sizes())
+        ? wrapIntegersInOptional(tensor.sym_sizes())
         : per_tensor_dynamic_dims_sizes[i];
-    std::vector<std::optional<int64_t>> tensor_dims_stride =
-        per_tensor_dynamic_dims_strides.empty()
-        ? wrapIntegersInOptional(tensor.strides())
+    std::vector<std::optional<c10::SymInt>> tensor_dims_stride = {};
+    tensor_dims_stride = per_tensor_dynamic_dims_strides.empty()
```
Comment (ezyang) [inline torch/csrc/dynamo/guards.cpp:278]: Why not assign it directly

**Review by ezyang — APPROVED** [review]
This is fine but please run perf test

**Comment by soulitzer** [comment]
Kicked off a workflow, from the dashboard we see no significant slowdowns:
image

File: torch/csrc/dynamo/guards.cpp (hunk 1 above)
Comment (soulitzer) [reply torch/csrc/dynamo/guards.cpp]: Fixed, I think this was from back when I special cased nested tensor still, but I forgot to remove.

**Comment by soulitzer** [comment]
image

**Pull request closed without merging**